internal/pkg/component/gitlab/gitlab.go

```diff
@@ -28,6 +30,31 @@ type Repository struct {
 	Repository   string
 }
 
+type FailedToParseUrlError struct {
+	url string
+	err string
+}
+
+func (e FailedToParseUrlError) Error() string {
+	return fmt.Sprintf("Failed to parse url: %s, error: %s", e.url, e.err)
+}
+
+type MissingSchemaError struct {
+	url string
+}
+
+func (e MissingSchemaError) Error() string {
+	return fmt.Sprintf("Failed to detect schema in url %s", e.url)
+}
+
+type MissingHostError struct {
+	url string
+}
+
+func (e MissingHostError) Error() string {
+	return fmt.Sprintf("Failed to detect host in url %s", e.url)
+}
+
```

Comment:
We don't need such checks, the url has been validated already while creating the component, such as: https://github.com/konflux-ci/mintmaker/blob/fc51f926095d3ba47b343d56a9d7951b4b035bee/internal/pkg/component/gitlab/gitlab.go#L33-L45